t/venom/pfservers/backup_db_and_restore/00_backup_db_and_restore.yml

```
@@ -0,0 +1,78 @@
+name: Backup db and restore 
+version: 2
+testcases:
+- name: create_users
+  steps:
+    - type: http
+      method: POST
+      url: '{{.pfserver_webadmin_url}}/api/v1/users'
+      ignore_verify_ssl: true
+      body: >-
+        {
+          "pid_overwrite": "0",
+          "pid": "{{.backup_db_and_restore.create_user}}",
+          "sponsor": "admin",
+          "valid_from": "{{.backup_db_and_restore.registration_user}}",
+          "expiration": "{{.backup_db_and_restore.expiration_user}}",
+          "actions": [
+            {
+              "type": "set_access_level",
+              "value": "NONE"
+            }
+          ]
+        }
+      headers:
+        "Authorization": "{{.pfserver_token}}"
+        "Content-Type": "application/json"
+      assertions:
+        - result.statuscode ShouldEqual 201
+
+- name: Backup
+  steps:
+  - script: /usr/local/pf/addons/backup-and-maintenance.sh
+
+- name: get_backup_name
+  steps:
+  - script: 'find /root/backup/packetfence-db-dump-*.sql.gz -mmin -5'
+    vars: 
+      backup_name:
+        from: result.systemout
+
+- name: unzip_db_backup
+  steps:
+  - script: 'gunzip {{.get_backup_name.backup_name}}'
+
+- name: remove_db_pf
+  steps:
+  - script: mysql -u {{.backup_db_and_restore.db_user}} -p{{.backup_db_and_restore.password_db_user}} -e "drop database pf;"
+
+- name: create_db_pf
+  steps:
+  - script: mysql -u {{.backup_db_and_restore.db_user}} -p{{.backup_db_and_restore.password_db_user}} -e "create database pf;"
+    assertions:
+    - result.code ShouldEqual 0
+
+- name: get_backup_name_uncompressed
+  steps:
+  - script: 'find /root/backup/packetfence-db-dump-*.sql -mmin -5'
+    vars:
+      backup_name_uncompressed:
+        from: result.systemout
+
+- name: inject_backup_db
+  steps:
+  - script: 'mysql -u {{.backup_db_and_restore.db_user}} -p{{.backup_db_and_restore.password_db_user}} pf < {{.get_backup_name_uncompressed.backup_name_uncompressed}}'
+    assertions:
+    - result.code ShouldEqual 0
+
+- name: Check_entry
+  steps:
+    - type: sql
+      driver: mysql
+      dsn: {{.backup_db_and_restore.db_user}}:{{.backup_db_and_restore.password_db_user}}@unix(/var/lib/mysql/mysql.sock)/pf
```

```suggestion
      dsn: root:{{.MYSQL_PWD}}@unix({{.MYSQL_UNIX_PORT}})/pf
```
To use environment variables.

---

After thinking a bit, it seems better to check if data injected at step 1 one is here using API in order to validate that PF is still able to access DB.